### PR TITLE
New plugin for parameter analysis

### LINKED ISSUE #491: Improve parameter distribution plugin

### PR DESCRIPTION
**Pull request by tnatt**
New plugin `ParameterAnalysis`. Dashboard to visualize parameter distributions and statistics for FMU ensembles, and to investigate parameter correlations on reservoir simulation time series data.
---

### Contributor checklist

- [x] :tada: This PR closes #491.
- [x] :scroll: I have broken down my PR into the following tasks:
   - [x] Add dashboard to visualize parameter distributions and compare parameter updates
   - [x] Add dashboard for parameter impact on simulation profiles
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [x] :book: I have considered adding a new entry in `CHANGELOG.md`, and added it if should be communicated there.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -68,6 +69,8 @@
         "pillow>=6.1",
         "pyscal>=0.4.1",
         "scipy>=1.2",
+        "scikit-learn>=0.23.1",
+        "statsmodels>=0.12.1",
```
Comment (anders-kiaer) [inline setup.py:73]: Comment that `statsmodels` is in practice an indirect dependency through https://plotly.com/python/linear-fits/?
How much work would it be to not add `scikit-learn` as a new dependency, and do the [normalization](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html) on our side?

File: webviz_subsurface/plugins/_parameter_analysis/figures/color_figure.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    "#ff1243",
+    "#be8091",
+    "#b2d4d7",
+    "#ff597b",
+    "#bdc3c7",
+    "#d8b2bd",
+    "#ffe7d6",
+    "#d5eaf4",
+    "#ff88a1",
+]
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_parameter_analysis/figures/color_figure.py:22]: Can we utilize the Equinor theme instead here?

File: webviz_subsurface/plugins/_parameter_analysis/models/ensemble_set_model.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+        ensemble_set_name: str = "EnsembleSet",
+        filter_file: Union[str, None] = "OK",
+    ) -> None:
+        self.ensemble_paths = ensemble_paths
+        self.ensemble_set_name = ensemble_set_name
+        self.filter_file = filter_file
+        self.webvizstore: List = []
+
+    def __repr__(self) -> str:
+        return f"EnsembleSetModel: {str(self.ensemble_paths)}"
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_parameter_analysis/models/ensemble_set_model.py:26]:
```suggestion
        return f"EnsembleSetModel: {self.ensemble_paths}"
```
Is this equivalent?

File: webviz_subsurface/plugins/_parameter_analysis/parameter_analysis.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+!> The `UNSMRY` files are auto-detected by `fmu-ensemble` in the `eclipse/model` folder of the \
+individual realizations. You should therefore not have more than one `UNSMRY` file in this \
+folder, to avoid risk of not extracting the right data.
+
+**Using aggregated data**
+
+?> Aggregated data may speed up the build of the app, as processing of `UNSMRY` files can be \
+slow for large models.
+
+"""
```
Comment (anders-kiaer) [inline webviz_subsurface/plugins/_parameter_analysis/parameter_analysis.py:69]: Docstrings can now be indented after https://github.com/equinor/webviz-config/pull/275.

**Review by anders-kiaer — COMMENTED** [review]
Awesome work @tnatt. 😎🥳  Don't have many comments besides @HansKallekleiv upcoming review. This will give new insight into FMU results.

image
Should we call it e.g. `Ensemble A` and `Ensemble B` instead of `Ensemble` and `Delta Ensemble`?

**Review by HansKallekleiv — APPROVED** [review]
🥳
